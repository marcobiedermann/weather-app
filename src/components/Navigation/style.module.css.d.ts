declare const styles: {
  readonly navigation: string;
  readonly navigation__list: string;
  readonly navigation__item: string;
  readonly navigation__link: string;
};
export = styles;
