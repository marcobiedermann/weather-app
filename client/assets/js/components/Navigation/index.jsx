import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';

const Navigation = props => {
  const { routes, ...otherProps } = props;

  return (
    <nav {...otherProps}>
      <ul>
        {routes.map(route => (
          <li key={route.path}>
            <Link to={route.path}>{route.name}</Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

Navigation.propTypes = {
  routes: PropTypes.arrayOf(PropTypes.shape()),
};

Navigation.defaultProps = {
  routes: [],
};

export default Navigation;
