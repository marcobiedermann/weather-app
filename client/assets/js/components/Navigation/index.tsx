import React, { FC } from 'react';
import { Link } from 'react-router-dom';

interface Route {
  name: string;
  path: string;
}

export interface NavigationProps {
  routes: Route[];
}

const Navigation: FC<NavigationProps> = (props) => {
  const { routes, ...otherProps } = props;

  return (
    <nav {...otherProps}>
      <ul>
        {routes.map((route) => (
          <li key={route.path}>
            <Link to={route.path}>{route.name}</Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navigation;
