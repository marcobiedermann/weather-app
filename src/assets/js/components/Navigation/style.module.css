/* stylelint-disable block-no-empty */

.navigation {
  text-transform: uppercase;
}

.navigation__list {
  list-style: none;
  margin-bottom: 1.5em;
  margin-top: 0;
  padding: 0;
}

.navigation__item {
}

.navigation__link {
  padding-bottom: 0.25em;
  padding-top: 0.25em;
}
