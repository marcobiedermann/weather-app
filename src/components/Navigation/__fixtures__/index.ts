const route = {
  name: 'Home',
  path: '/',
};

const routes = [route];

export default routes;
