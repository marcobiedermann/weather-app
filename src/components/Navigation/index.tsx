import clsx from 'clsx';
import { ReactNode } from 'react';
import { NavLink } from 'react-router-dom';
import styles from './style.module.css';

interface Route {
  icon?: ReactNode;
  name: string;
  path: string;
}

interface NavigationProps {
  routes: Route[];
}

function Navigation(props: NavigationProps): JSX.Element {
  const { routes, ...otherProps } = props;

  return (
    <nav className={styles.navigation} {...otherProps}>
      {routes.length > 0 && (
        <ul className={styles.navigation__list}>
          {routes.map((route) => {
            const { icon, name, path } = route;

            return (
              <li key={path} className={styles.navigation__item}>
                <NavLink
                  to={path}
                  className={({ isActive }) =>
                    clsx(styles.navigation__link, {
                      [styles.navigation__linkActive]: isActive,
                    })
                  }
                >
                  {icon}
                  <span>{name}</span>
                </NavLink>
              </li>
            );
          })}
        </ul>
      )}
    </nav>
  );
}

export type { NavigationProps };
export default Navigation;
