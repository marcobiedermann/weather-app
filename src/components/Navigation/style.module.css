/* stylelint-disable block-no-empty */

.navigation {
}

.navigation__list {
  list-style: none;
  margin: 0;
  padding: 0;
}

.navigation__item {
}

.navigation__link {
  align-items: center;
  display: flex;
  gap: 0.5em;
  padding: 0.5em 3em;
  position: relative;
  text-decoration: none;
}

.navigation__linkActive::after {
  background-color: currentcolor;
  content: '';
  height: 100%;
  position: absolute;
  right: 0;
  width: 3px
}