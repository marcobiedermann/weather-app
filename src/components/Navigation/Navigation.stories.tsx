import type { Meta, StoryObj } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';
import Navigation from '.';

const meta: Meta<typeof Navigation> = {
  component: Navigation,
  args: {
    routes: [
      {
        name: 'Home',
        path: '/',
      },
    ],
  },
  decorators: [
    (Story) => (
      <MemoryRouter>
        <Story />
      </MemoryRouter>
    ),
  ],
};

type Story = StoryObj<typeof Navigation>;

const Default: Story = {};

export { Default };
export default meta;
