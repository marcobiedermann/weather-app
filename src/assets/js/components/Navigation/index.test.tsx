import React from 'react';
import renderer from 'react-test-renderer';
import Navigation from '.';

describe('components/Navigation', () => {
  it('renders correctly', () => {
    expect.assertions(1);

    const tree = renderer.create(<Navigation routes={[]} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
