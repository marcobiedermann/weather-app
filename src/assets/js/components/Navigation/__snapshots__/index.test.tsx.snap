// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navigation renders correctly 1`] = `
<nav
  className="navigation"
>
  <ul
    className="navigation__list"
  />
</nav>
`;
